code/salary-calculator/ViewModel/src/test/java/ru/unn/agile/salarycalculator/viewmodel/legacy/ViewModelTests.java

```diff
@@ -0,0 +1,160 @@
+package ru.unn.agile.salarycalculator.viewmodel.legacy;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import ru.unn.agile.salarycalculator.viewmodel.legacy.ViewModel.Status;
+
+import static org.junit.Assert.*;
+
+public class ViewModelTests {
+    private ViewModel viewModel;
+
+    @Before
+    public void setUpEmptyExample() {
+        viewModel = new ViewModel();
+        viewModel.setSalary("10000");
+        viewModel.setWorkedHours("154");
+        viewModel.setCountMonth("10");
+        viewModel.setCountYear("2014");
+    }
+
+    @After
+    public void deleteExample() {
+        viewModel = null;
+    }
+
+    @Test
+    public void checkStatusInBegin() {
+        viewModel = new ViewModel();
+        assertEquals(Status.COUNT_WAITING, viewModel.getStatus());
+    }
+
+    @Test
+    public void checkSetters() {
+        viewModel.setSalary("10000");
+        viewModel.setWorkedHours("145");
+        viewModel.setCountMonth("5");
+        viewModel.setCountYear("2000");
+
+        assertEquals("10000", viewModel.getSalary());
+        assertEquals("145", viewModel.getWorkedHours());
+        assertEquals("5", viewModel.getCountMonth());
+        assertEquals("2000", viewModel.getCountYear());
+    }
+
+    @Test
+    public void checkStatusWhenReadyCalculate() {
+        viewModel.checkCountFields();
+
+        assertEquals(Status.READY_CALCULATE, viewModel.getStatus());
+    }
+
+    @Test
+    public void checkStatusCash() {
+        viewModel.checkCountFields();
+        viewModel.calculateSalary();
+
+        assertEquals(Status.CASH, viewModel.getStatus());
+    }
+
+    @Test
+    public void isLengthCharactersSalaryNotCorrect() {
+        viewModel.setSalary("1000000000000000000000000");
```

Comment:
I'm not sure, that it's bad format :)

Reply:
based on offline discussion with Vitaly, it's ok to set such big numbers because only one limitation is size of integer format